### ISSUE TITLE
Css source maps empty when doing roc build

### ISSUE THREAD

**Issue by inside** [rocjs/roc#150]
Hi,

I followed the steps on http://www.getroc.org/ and noticed after `roc build` css source maps are empty. For example:

```
% cat build/client/app.09aad10db2d7d2134ea1.css.map 
{"version":3,"sources":[],"names":[],"mappings":"","file":"app.09aad10db2d7d2134ea1.css","sourceRoot":""}
```

But the css is built correctly:

```
% more build/client/app.09aad10db2d7d2134ea1.css
*{box-sizing:border-box;margin:0;padding:0}._3H4mn{height:50px;background:#fff}._3H4mn ul{list-style:inside;list-style-type:none}._3H4mn li{display:inline-block;margin-right:30px}._3wvJ
G{border:10px solid pink}._1X9AS{padding:5px;line-height:20px;color:#1d7c8e;text-decoration:none;font-family:Rajdhani;font-size:20px;font-weight:500;text-transform:uppercase}.hamBV{bord
er-bottom:2px solid #1d7c8e}._1X9AS:visited{color:#1d7c8e}._1X9AS:hover{color:#167688}body{background:#fff;font-family:Helvetica Neue,helvetica,sans-serif;font-size:16px;font-weight:300
}h1{font-size:90px;font-weight:800}h1,h2{font-family:Arial;text-align:center}h2{font-size:35px;font-weight:400;text-transform:uppercase;line-height:36px}.Wg7ma{border:1px solid red}h3{f
ont-family:Arial;font-size:36px;font-weight:400;text-align:center;text-transform:uppercase;margin-bottom:30px}h4{font-size:24px}h4,h5{margin-top:20px;font-family:Arial;font-weight:500}h
5{font-size:20px}pre{margin-top:10px;padding:16px;overflow:auto;font-size:85%;line-height:1.45;background-color:#f3f3f3;border-radius:3px}._3wEwY{margin:auto}a:visited{color:red}p{margi
n-bottom:7px}.header{height:180px;text-align:center;color:#fff;background:#2a7e8e}.header h1{text-transform:uppercase}.wrapper{padding:10px;width:100%;margin:auto}@media (min-width:800p
x){.wrapper{width:800px;margin:auto}}@media (max-width:700px){h1{font-size:70px}}@media (max-width:500px){h1{font-size:40px}}._1oth-{margin-top:20px;height:90px;width:300px;color:#2a7e8
e;font-size:28px;border:1px solid #fff;padding:20px}._1oth- input{float:left;margin-left:20px;margin-top:10px;width:50px;padding:2px}._1oth- button{float:left;margin-left:20px;padding:4
px;height:50px;width:80px}.gw0f0{float:left}._1QYX8{background:#dcebee;color:#000}._1QYX8 ul{list-style-position:inside;list-style-type:disc}._2vF0H{margin:auto;background:#dcebee;color
:#000}
/*# sourceMappingURL=app.09aad10db2d7d2134ea1.css.map*/
```

This makes it pretty hard to debug css on production.

Can you have a look at this or point me to the right direction so that I can propose a pull request?

Thanks

**Comment by dlmr** [comment]
Hello!

I'm assuming that you are using version `1.0.0-rc.11` or higher of Roc, is that correct? 

SourceMaps for CSS are little buggy in general as can be read on the `css-loader` documentation:

> They are not enabled by default because they expose a runtime overhead and increase in bundle size (JS SourceMap do not). In addition to that relative paths are buggy and you need to use an absolute public path which include the server url.
> 
> https://github.com/webpack/css-loader#sourcemaps

We actually have disabled them from being generated for now in the latest versions because of this, [as done here](https://github.com/rocjs/roc-plugin-style-css/pull/8/commits/4a48779e4091be28ef6a29a03d01116b0601ca10). The `.css.map` file should really not be generated at all here, that is a bug.

In general I found that they actually do not bring that much value to the application since the styles will be nicely formatted in the browsers devtools. However, and maybe the thing that you are missing the most, is that the classes are changed when using CSS Modules which do make it a little hard to understand from where a certain rule is coming. 

I think enabling them again would be fantastic as long as we make it an opt-in and make sure that they work as expected when enabled. A great place to start experimenting with this is to look at the [`roc-plugin-style-css`](https://github.com/rocjs/roc-plugin-style-css) plugin as referenced above.

**Comment by inside** [comment]
Hi Gustaf,

> I'm assuming that you are using version 1.0.0-rc.11 or higher of Roc, is that correct?

Correct:

```
% roc --version
1.0.0-rc.14
```

I understand the overhead caused by css sourcemaps and deciding to disable them. The "empty" `.css.map` is generated when the webpack `devtool` key config is set for example to `'source-map'`. Bug or feature, I don't know.

> However, and maybe the thing that you are missing the most, is that the classes are changed when using CSS Modules which do make it a little hard to understand from where a certain rule is coming.

What you describe is exactly my concern. With sourcemaps and CSS Modules, I can pinpoint the css rules. Without sourcemaps how could we have this convinience?

I had a look at the roc-plugin-style-css and would be glad to make the sourcemaps as an opt-in. IMHO, this opt-in option should live in the `roc.config.js` file. Correct me if I'm wrong, I still have to grasp the roc architecture.

Regards,

**Comment by dlmr** [comment]
> Without sourcemaps how could we have this convinience?

One option could be to make it possible to define how the CSS Modules names should be defined, allowing to keep the original name together with the hash. This would increase the amount of data downloaded to the user slightly since the strings would be longer but you would be able to know what the original class was. This is basically what we are doing in development currently.

> I had a look at the roc-plugin-style-css and would be glad to make the sourcemaps as an opt-in. IMHO, this opt-in option should live in the roc.config.js file. Correct me if I'm wrong, I still have to grasp the roc architecture.

Yes, adding as an option to the `roc.config.js` file is the correct approach.

**Comment by dlmr** [comment]
Will close this now since this have been implemented in `roc-plugin-style-css@1.0.0-beta.5`. 🙂

**Comment by inside** [comment]
:+1:
